packages/yew/src/dom_bundle/bsuspense.rs

```diff
@@ -163,22 +203,89 @@ impl Reconcilable for VSuspense {
                 fallback_ref
             }
             // Freshly unsuspended. Detach fallback from the DOM, then shift children into it.
             (false, Some(_)) => {
-                suspense
-                    .fallback_bundle
-                    .take()
-                    .unwrap() // We just matched Some(_)
-                    .detach(root, parent, false);
+                match suspense.fallback.take() {
+                    Some(Fallback::Bundle(bundle)) => {
+                        bundle.detach(root, parent, false);
+                    }
+                    #[cfg(feature = "hydration")]
+                    Some(Fallback::Fragment(fragment)) => {
+                        fragment.detach(root, parent, false);
+                    }
+                    None => {
+                        unreachable!()
+                    }
+                };
 
                 children_bundle.shift(parent, next_sibling.clone());
                 children.reconcile_node(root, parent_scope, parent, next_sibling, children_bundle)
             }
         }
     }
 }
+
+#[cfg(feature = "hydration")]
+mod feat_hydration {
+    use super::*;
+
+    use crate::dom_bundle::{Fragment, Hydratable};
+    use crate::virtual_dom::Collectable;
+
+    impl Hydratable for VSuspense {
+        fn hydrate(
+            self,
+            root: &BSubtree,
+            parent_scope: &AnyScope,
+            parent: &Element,
+            fragment: &mut Fragment,
+        ) -> (NodeRef, Self::Bundle) {
+            let detached_parent = document()
+                .create_element("div")
+                .expect("failed to create detached element");
+
+            let collectable = Collectable::Suspense;
+            let fallback_fragment = Fragment::collect_between(fragment, &collectable, parent);
+
+            let mut nodes = fallback_fragment.deep_clone();
+
+            for node in nodes.iter() {
+                detached_parent.append_child(node).unwrap();
+            }
+
+            let (_, children_bundle) =
+                self.children
+                    .hydrate(root, parent_scope, &detached_parent, &mut nodes);
+
+            // We trim all leading text nodes before checking as it's likely these are whitespaces.
+            nodes.trim_start_text_nodes(&detached_parent);
+
+            assert!(nodes.is_empty(), "expected end of suspense, found node.");
+
+            let node_ref = fallback_fragment
+                .front()
+                .cloned()
+                .map(NodeRef::new)
+                .unwrap_or_default();
+
+            (
+                node_ref,
+                BSuspense {
+                    children_bundle,
+                    detached_parent,
+                    key: self.key,
+
+                    // We start hydration with the BSuspense being suspended.
+                    // A subsequent render will resume the BSuspense if not needed to be suspended.
```

Comment:
```
<Suspension fallback={"not relevant here"}>
  <input {onclick} />
</Suspension>
```

will never unsuspend, and worse, the hydration taking place on the hidden detached parent means the listener is never installed. How about just hydrating the components as normal, in place in the dom, and shift them to the detached parent if one of them decides to suspend (and render the fallback as normal)? I don't think it's necessary to clone and populate before that happens. The fallback is never rendered on the server anyway.

Reply:
> will never unsuspend, and worse, the hydration taking place on the hidden detached parent means the listener is never installed.

There's a test case that checks it.

> How about just hydrating the components as normal, in place in the dom, and shift them to the detached parent if one of them decides to suspend (and render the fallback as normal)?

This is not the expected behaviour. This would cause the main content to flash before it is switch to fallback.